core.h

```diff
@@ -15,6 +15,11 @@ typedef struct ByteArray {
     int length;
 } ByteArray;
 
+typedef struct DataPtr {
```

Comment:
See previous comment, seems like could just use the ByteArray type instead of adding an additional type.